clean up Flyte documentation (#1102)
flyteorg/flyte: rsts/concepts/basics.rst

.. _divedeep:

#############
Core Concepts
#############

.. panels::
    :header: text-center
 
    .. link-button:: divedeep-tasks
       :type: ref
       :text: Tasks
       :classes: btn-block stretched-link
    ^^^^^^^^^^^^
    A **Task** is any independent unit of processing. They can be pure functions or functions with side-effects.

>     They also have configuration and requirements specification associated with each definition of the Task.

Reviewer: ```suggestion
    Each definition of a task also has associated configuration and requirements specifications.
```

Author: as part of the RFC: https://docs.google.com/document/d/1Yp5cYujKT6UMv17Y6r1djMchaMZWAlPNUTZnbw1sZvc/edit# we've moved all of the "how to" articles to the new "User Guide" under the appropriate topic. This is to avoid duplicate information in the documentation, and moving forward the plan is to write user guide articles and use the [discussion page](https://github.com/flyteorg/flyte/discussions) to point users to the user guide page that addresses the question
